### PR TITLE
(SIMP-2053) New Contributors Guide

### PR DESCRIPTION
**Pull request by jeefberkey**
We have moved our code review process to Github, and the old
Contributors guide needed to be updated.

SIMP-2053 #close

---
This change is [Reviewable](https://reviewable.io/reviews/simp/simp-doc/129)

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[docs/contributors_guide/Contribution_Procedure.rst, line 67 at r3](https://reviewable.io:443/reviews/simp/simp-doc/129#-KXCbuiyhw6QBnBoIMzp:-KXCbuizbKTGn6trBJyL:b-wpud6x) ([raw file](https://github.com/simp/simp-doc/blob/164c46f97e02c627b8bfdb52a2b0549643d1d515/docs/contributors_guide/Contribution_Procedure.rst#L67)):*
> ```reStructuredText
>      project's `HipChat`_ channel
> 
> #. After passing Travis-CI tests, the GitHub pull request must be reviewed by a
> ```

The 'a' at the end of this line should be removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/simp/simp-doc/129)*

**Comment by jeefberkey** [comment]
@lamawithonel I added a shorter section with hub to emphasize some of it's features

**Comment by lamawithonel** [comment]
Reviewed 2 of 3 files at r1, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[docs/contributors_guide/Contribution_Procedure.rst, line 1 at r4](https://reviewable.io:443/reviews/simp/simp-doc/129#-KXHCXMdzRXaMyy0Kfb4:-KXHCT7Jm44tVUYMjJsr:b-e3ittl) ([raw file](https://github.com/simp/simp-doc/blob/98533eb2ae6b41e44faf19b401187582eef289d7/docs/contributors_guide/Contribution_Procedure.rst#L1)):*
> Contribution Procedure (Using hub)

Move the section to the bottom and I'll be OK with it. It's OK to recommend it and link to the section in the first paragraph, but as a contributor guide, the first section should be the bare bones, absolute minimum requirements for contribution. I know hub is kind of awesome, but it's really secondary to the contribution process. You can leave the long version as the default for now and I'll come back to clean it up next week.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/simp/simp-doc/129)*

**Comment by trevor-vaughan** [comment]
@jeefberkey @nick-markowski and @lamawithonel Please review the latest change set that I added.

**Comment by jeefberkey** [comment]
I like the new changes and I think they are definitely using a better approach than my first draft

File: docs/contributors_guide/Contribution_Procedure.rst [hunk 1]
```diff
@@ -1,50 +1,108 @@
...
+    * Followed by a line of white space
 
-     * First line:
+  * Subsequent lines:
 
-       * Start with the Issue name in parentheses [e.g., ``SIMP-999``],
-         followed by a summary of the change
-       * No longer than **50** characters
-       * Followed by a line of white space
+    * Each line should be no longer than **72** characters
```
Comment (lamawithonel) [inline docs/contributors_guide/Contribution_Procedure.rst]: This should also be expanded to cover the substance of the message. "Use the body to explain what and why vs. how.," as Chris Beams puts it. Puppet has a good example [here](https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md).  @op-ct summed it up something like this:

> `<previous behavior> <why that is a problem> <details of what and how the change fixes the problem>`

File: docs/contributors_guide/Contribution_Procedure.rst [hunk 2]
```diff
@@ -1,50 +1,108 @@
...
 
-       * Each line should be no longer than **72** characters
+    * `JIRA issues can be referenced`_ at the end of the commit message
+    * It is recommended to only use `JIRA Smart Commit Tags`_ ``#comment`` and
+      ``#close``
+    * Avoid ``#resolve`` and ``#time`` as it will not update JIRA until
+      after the issue is merged
 
-     * Issue references:
+Contribution Procedure (Long Version)
```
Comment (lamawithonel) [inline docs/contributors_guide/Contribution_Procedure.rst]: I'm not sure I understand the value of this section when the tl;dr above is already quite long.  IMO, we should include one section that includes every step in detail, and a true tl;dr overview that is small enough to fit on a standard VT100 terminal, i.e., no more than 80x25 characters in size.

File: docs/contributors_guide/Contribution_Procedure.rst [hunk 3]
```diff
@@ -55,53 +113,67 @@ Contribution Procedure
      * Use ``git rebase -i`` to squash commits (use ``git rebase -i HEAD~n`` to
        go back ``n`` commits)
 
+   * Make sure to select the option `Allow edits from maintainers`.  This
+     will allow people in the organization to make edits to the pull request.
```
Comment (lamawithonel) [inline docs/contributors_guide/Contribution_Procedure.rst]: Some people might not want others editing their commits. In that case, it's worth saying something like, "or if you don't want us messing with your commit, be attentive to requests for changes."

File: docs/contributors_guide/Contribution_Procedure.rst (hunk 2 above)
Comment (op-ct) [inline docs/contributors_guide/Contribution_Procedure.rst]: I agree with @lamawithonel about this―the initial section was originally intended to be a quickstart overview and would scan much more easily to new contributors if it were merely the steps' summaries + the commit message format, with links to the expanded procedure where needed.  Certainly, steps like `cd iptables` are overkill in an overview.

For a document intended to be rendered from reStructuredText and viewed online, I am not convinced an arbitrary limitation such as VT100 dimensions is appropriate, though.

File: docs/contributors_guide/Contribution_Procedure.rst [hunk 4]
```diff
@@ -55,53 +113,67 @@ Contribution Procedure
...
 #. `Travis-CI`_ will notice the pull request and run CI tests
 
    * Travis-CI will run tests based on the `.travis.yml`_ file in the
      repository's top-level directory
-   * Travis-CI results can be seen in the PR on GitHub and are posted to the
-     project's `HipChat`_ channel
+   * Travis-CI results can be seen in the pull request on GitHub and are posted
+     to the project's `HipChat`_ channel
+
+#. After passing Travis-CI tests, the GitHub pull request must be reviewed by a
```
Comment (op-ct) [inline docs/contributors_guide/Contribution_Procedure.rst]: I haven't seen any documentation on the review process itself.  Perhaps a link to something like https://help.github.com/articles/reviewing-proposed-changes-in-a-pull-request/ would be enough.

File: docs/contributors_guide/Contribution_Procedure.rst (hunk 2 above)
Comment (lamawithonel) [inline docs/contributors_guide/Contribution_Procedure.rst]: My suggestion for VT100 dimensions isn't so much for screen size constraints as much as a measure of simplicity.

**Pull request merged**